src/main/resources/native/android/dalvik/MainActivity.java

```diff
@@ -228,5 +255,239 @@ protected void onStop() {
 
+    public final static int PRESS   = 111;
+    public final static int RELEASE = 112;
+    public final static int TYPED   = 113; 
+
+    private int deadKey = 0;
+
+    void processAndroidKeyEvent (KeyEvent event) {
+        System.out.println("KeyEvent: " + event+" with action = "+event.getAction());
+        int jfxModifiers = mapAndroidModifierToJfx(event.getMetaState());
+        switch (event.getAction()) {
+            case KeyEvent.ACTION_DOWN:
+                KeyCode jfxKeyCode = mapAndroidKeyCodeToJfx(event.getKeyCode());
+System.out.println ("[JVDBG] eventkeycode = "+event.getKeyCode()+" and jfxkc = "+jfxKeyCode+" with code "+ jfxKeyCode.impl_getCode());
+                nativedispatchKeyEvent(PRESS, jfxKeyCode.impl_getCode(), jfxKeyCode.impl_getChar().toCharArray(), jfxKeyCode.impl_getChar().toCharArray().length, jfxModifiers);
+                break;
+
+            case KeyEvent.ACTION_UP:
+                jfxKeyCode = mapAndroidKeyCodeToJfx(event.getKeyCode());
+                nativedispatchKeyEvent(RELEASE, jfxKeyCode.impl_getCode(), jfxKeyCode.impl_getChar().toCharArray(), jfxKeyCode.impl_getChar().toCharArray().length, jfxModifiers);
+                int unicodeChar = event.getUnicodeChar();
+                if ((unicodeChar & KeyCharacterMap.COMBINING_ACCENT) != 0) {
+                    deadKey = unicodeChar & KeyCharacterMap.COMBINING_ACCENT_MASK;
+                    return;
+                }
+
+                if (deadKey != 0 && unicodeChar != 0) {
+                    unicodeChar = KeyCharacterMap.getDeadChar(deadKey, unicodeChar);
+                    deadKey = 0;
+                }
+
+                if (unicodeChar != 0) {
+                    nativedispatchKeyEvent(TYPED, KeyCode.UNDEFINED.impl_getCode(), Character.toChars(unicodeChar), 1, jfxModifiers);
+                }
+
+                break;
+
+            case KeyEvent.ACTION_MULTIPLE:
+                if (event.getKeyCode() == KeyEvent.KEYCODE_UNKNOWN) {
+                    nativedispatchKeyEvent(TYPED, KeyCode.UNDEFINED.impl_getCode(), event.getCharacters().toCharArray(), event.getCharacters().toCharArray().length,    jfxModifiers);
+                } else {
+                    jfxKeyCode = mapAndroidKeyCodeToJfx(event.getKeyCode());
+                    for (int i = 0; i < event.getRepeatCount(); i++) {
+                        nativedispatchKeyEvent(PRESS, jfxKeyCode.impl_getCode(), null, 0, jfxModifiers);
+                        nativedispatchKeyEvent(RELEASE, jfxKeyCode.impl_getCode(), null, 0, jfxModifiers);
+                        nativedispatchKeyEvent(TYPED, jfxKeyCode.impl_getCode(), null, 0, jfxModifiers);
+                    }
+                }
+
+                break;
+            default:
+                System.err.println("DalvikInput.onKeyEvent Unknown Action " + event.getAction());
+                break;
+        }
+    }
+
+    private final static int MODIFIER_SHIFT = 1;
+    private final static int MODIFIER_CONTROL = 2;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4825537)